ori/volume/server/volume.go

```diff
@@ -16,235 +16,143 @@ package server
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
 	"strings"
+	"time"
 
+	"github.com/go-logr/logr"
 	"github.com/google/uuid"
-	"github.com/onmetal/cephlet/ori/volume/apiutils"
-	"github.com/onmetal/cephlet/pkg/ceph"
-	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+	orimeta "github.com/onmetal/onmetal-api/ori/apis/meta/v1alpha1"
 	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	userIDKey  = "userID"
-	userKeyKey = "userKey"
-
-	pvPoolKey               = "pool"
-	monitorConfigMapDataKey = "csi-cluster-config-json"
-	pvImageNameKey          = "imageName"
-	pvRadosNamespaceKey     = "radosNamespace"
+	// worldwide number key
+	// to use WWN Company Identifiers, set wwnPrefix to Private "1100AA"
+	wwnPrefix string = ""
+
+	MonitorsKey = "monitors"
+	ImageKey    = "image"
+	UserIDKey   = "userID"
+	UserKeyKey  = "userKey"
 )
 
-type AggregateCephVolume struct {
-	Pvc          *corev1.PersistentVolumeClaim
-	ImagePvc     *corev1.PersistentVolumeClaim
-	VolumeClass  *storagev1alpha1.VolumeClass
-	AccessSecret *corev1.Secret
-}
-
-var onmetalVolumeStateToORIState = map[corev1.PersistentVolumeClaimPhase]ori.VolumeState{
-	corev1.ClaimPending: ori.VolumeState_VOLUME_PENDING,
-	corev1.ClaimBound:   ori.VolumeState_VOLUME_AVAILABLE,
-	corev1.ClaimLost:    ori.VolumeState_VOLUME_ERROR,
+type AggregateVolume struct {
+	Requested   Volume
+	Provisioned Image
 }
 
-func generateWWN(wwnPrefix string) (string, error) {
-	// prefix is optional, set to 1100AA for private identifier
-	wwn := wwnPrefix
+type Volume struct {
+	Name string
 
-	// use UUIDv4, because this will generate good random string
-	wwnUUID, err := uuid.NewRandom()
-	if err != nil {
-		return "", fmt.Errorf("failed to generate UUIDv4 for WWN: %w", err)
-	}
+	Bytes uint64
 
-	// append hex string without "-"
-	wwn += strings.Replace(wwnUUID.String(), "-", "", -1)
+	Class string
+	IOPS  int64
+	TPS   int64
 
-	// WWN is 64Bit number as hex, so only the first 16 chars are returned
-	return wwn[:16], nil
+	Image *PopulationImage
 }
 
-func (s *Server) convertOnmetalVolumeState(state corev1.PersistentVolumeClaimPhase) (ori.VolumeState, error) {
-	if state, ok := onmetalVolumeStateToORIState[state]; ok {
-		return state, nil
-	}
-	return 0, fmt.Errorf("unknown onmetal volume state %q", state)
+type PopulationImage struct {
```

Comment:
Something about the naming here `PopulationImage` / `Image` seems a bit off - let's discuss about this.